core/src/test/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateActionTests.java

```diff
@@ -658,8 +670,17 @@ public void testInterleavedPublishCommit() throws Throwable {
         final ClusterState finalState = states.get(numOfStates - 1);
         Collections.shuffle(states, random());
 
-        logger.info("--> publishing states");
+        List<ClusterState> orderedSubsequence = new ArrayList<>();
```

Comment:
why do we need the subsequence? the original list is already ordered - we can just use it? also note that strictly speaking we do support out of order publishing (until we remove the settings that allows it, one day.. :)).